include/loader/loader.h

```diff
@@ -152,30 +152,50 @@ class Loader {
   }
   template <typename T, typename L>
   Expect<void> loadSectionContentVec(T &Sec, L &&Func) {
-    uint32_t VecCnt = 0;
+    if (auto Res = loadVec(Sec.getContent(), std::move(Func)); !Res) {
+      return logLoadError(Res.error(), FMgr.getLastOffset(),
+                          NodeAttrFromAST<T>());
+    }
+    return {};
+  }
+
+  Expect<uint32_t> loadVecCnt() {
     // Read the vector size.
     if (auto Res = FMgr.readU32()) {
-      VecCnt = *Res;
-      if (VecCnt / 2 > FMgr.getRemainSize()) {
-        return logLoadError(ErrCode::Value::IntegerTooLong,
-                            FMgr.getLastOffset(), NodeAttrFromAST<T>());
+      // Question: why we used to divide by 2 here?
```

Comment:
According to the definition, there should not be larger than 2*remainsize. Otherwise, we should show an error about the length being too long.

Reply:
I have added these explanation to where the commented code is.

Reply:
I might use
```
// NOTE: why ...?
//
// According to ...
```
to avoid confusion here for reviewer